### ISSUE TITLE
failed to map merge request event string because of date format parse error

### ISSUE THREAD

**Issue by lethe3000** [gmessner/gitlab4j-api#151]
gitlab version: 10.5.1
api version: 4.8.5

code as below
```
ObjectMapper objectMapper = new ObjectMapper(); 
objectMapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE); 
objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false); 
return objectMapper.readValue(content, Event.class);
```

error message:
```
com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `java.util.Date` from String "2018-03-07 19:52:55 +0800": not a valid representation (error: Failed to parse Date value '2018-03-07 19:52:55 +0800': Cannot parse date "2018-03-07 19:52:55 +0800": while it seems to fit format 'yyyy-MM-dd'T'HH:mm:ss.SSSZ', parsing fails (leniency? null))

```

event string(with some fields omitted)
```
{
  "object_kind": "merge_request",
  "user": { },
  "project": {},
  "object_attributes": {
    "created_at": "2018-03-07 19:52:55 +0800",
    "updated_at": "2018-03-07 19:52:55 +0800"
  }
}
```

**Comment by gmessner** [comment]
@lethe3000 
The classes in this project are set up to use an ObjectMapper created by this project, it makes creating the entity classes very simple.  You should use the following to get an ObjectMapper:
```
import org.gitlab4j.api.utils.JacksonJson;

JacksonJson jacksonJson = new JacksonJson();
ObjectMapper objectMapper = jacksonJson.getObjectMapper();
```

Or if you must use your own ObjectMapper, look at how JacksonJson constructs one and replicate its construction.

Also, there are convenience methods in JacksonJson that you can use to create objects, don't even need the ObjectMapper.

**Comment by lethe3000** [comment]
verified, close this issue

**Comment by gmessner** [comment]
@lethe3000 
Wanted to mention that GitLab4J also has built-in support for handling both Webhooks and System Hooks, see the following classes for more information:
```
org.gitlab4j.api.webhook.WebHookManager
org.gitlab4j.api.systemhooks.SystemHookManager
```

**Comment by lethe3000** [comment]
I use event handler code from WebHookManager to handle MergeRequest event, still failed

```
org.gitlab4j.api.webhook.Event event;
String postData = HttpRequestUtils.getPostDataAsString(request);
JacksonJson jacksonJson = new JacksonJson();
event = jacksonJson.unmarshal(org.gitlab4j.api.webhook.Event.class, postData);
logger.info(event);
```

merge request created event:
```
com.fasterxml.jackson.databind.JsonMappingException: 2018-03-12 10:16:46 +0800 (through reference chain: org.gitlab4j.api.webhook.MergeRequestEvent["object_attributes"]->org.gitlab4j.api.webhook.MergeRequestEvent$ObjectAttributes["created_at"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:391)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:351)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1689)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:289)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:287)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:189)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:161)
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer._deserializeTypedForId(AsPropertyTypeDeserializer.java:130)
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromObject(AsPropertyTypeDeserializer.java:97)
	at com.fasterxml.jackson.databind.deser.AbstractDeserializer.deserializeWithType(AbstractDeserializer.java:254)
	at com.fasterxml.jackson.databind.deser.impl.TypeWrappedDeserializer.deserialize(TypeWrappedDeserializer.java:68)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4001)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2992)
	at org.gitlab4j.api.utils.JacksonJson.unmarshal(JacksonJson.java:117)
	at com.wlqq.dms.api.event.controller.EventController.gitlabHook1(EventController.java:78)
```

event data from gitlab
```
{
  "object_kind": "merge_request",
  "user": {...},
  "project": {...},
  "object_attributes": {
    "created_at": "2018-03-12 10:16:46 +0800",
    "description": "",
    "head_pipeline_id": null,
    "merge_params": {
      "force_remove_source_branch": "0"
    },
    "merge_status": "unchecked",
    "merge_user_id": null,
    "merge_when_pipeline_succeeds": false,
    "milestone_id": null,
    "state": "opened",
    "target_branch": "master",
    "time_estimate": 0,
    "title": "feat(INF_DMS-1014): 011",
    "updated_at": "2018-03-12 10:16:46 +0800",
    "updated_by_id": null,
    "source": {...},
    "target": {...},
    "last_commit": {
      "timestamp": "2018-03-12T10:16:09+08:00"
    },
    "work_in_progress": false,
    "total_time_spent": 0,
    "human_total_time_spent": null,
    "human_time_estimate": null,
    "action": "open"
  },
  "labels": [
  ],
  "changes": {...}
}
```
I think the problem is to handle object_attributes.created_at and updated_at time format

**Comment by gmessner** [comment]
@lethe3000 
You are correct, both created_at and updated_at are invalid ISO8601 formats.  I will add handling of this to the ISO8601 class and release with the next release, which is likely tomorrow.

**Comment by gmessner** [comment]
@lethe3000 
Are you getting this data from a gitlab server, the GitLab docs at the following URL do not show the format used by created_at and updated_at.

https://docs.gitlab.com/ce/user/project/integrations/webhooks.html#merge-request-events

**Comment by lethe3000** [comment]
Yes my gitlab server version is latest 10.5.1
and the data can be view from project settings -> integrations -> webhook item's "edit" button -> Recent Deliveries list
![image](https://user-images.githubusercontent.com/776658/37268166-9465e692-25ff-11e8-9741-cf07977e94e1.png)

**Comment by gmessner** [comment]
@lethe3000 
I've just never seen that format before, and it is not valid XSD or ISO8601.  That being said I have added handling of them. The PR should likely be merged and released tomorrow.

**Comment by lethe3000** [comment]
maybe you can try to recurrent this issue by set your gitlab server to different timezone?
thanks very much to help me resolve this issue

**Comment by gmessner** [comment]
@lethe3000 
I just released 4.8.6 with the fix in it.  If you could verify and close this issue I would appreciate it.

**Comment by lethe3000** [comment]
4.8.6 works for me, thank you
